### PR TITLE
add initJob.podSpec

### LINKED ISSUE #297: [FR]: activeDeadlineSeconds
It is useful to outfit the [job-init](https://github.com/zammad/zammad-helm/blob/main/zammad/templates/job-init.yaml) with high ressources. Especially the [Elastic Search init](https://github.com/zammad/zammad-helm/blob/main/zammad/templates/configmap-init.yaml#L35) needs a huge amount of memory.

So I have to buy this memory, only for these rare init occasions. But my cluster offers me an additional [Terminating quota-scope](https://kubernetes.io/docs/concepts/policy/resource-quotas/#quota-scopes).
To get this Terminating quota-scope, I need to add an [activeDeadlineSeconds](https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-termination-and-cleanup) to the job-init.

### PR DESCRIPTION
**Pull request by klml**
#### What this PR does / why we need it

Adds podSpec to the [initJob](https://github.com/zammad/zammad-helm/blob/main/zammad/templates/job-init.yaml)

#### Which issue this PR fixes

With this PR I can add
 
```
  initJob:
    podSpec:
      activeDeadlineSeconds: 1337
```
and fix #297

#### Checklist

- [x] Chart Version bumped

### CODE REVIEW

File: zammad/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: zammad
-version: 12.4.1
+version: 12.4.2
```
Comment (monotek) [inline zammad/Chart.yaml:3]: new feature should raise minor version

File: zammad/templates/job-init.yaml [hunk 2]
```diff
@@ -35,6 +35,9 @@ spec:
         {{- end }}
     spec:
       {{- include "zammad.podSpec" . | nindent 6 }}
+      {{- with .Values.zammadConfig.initJob.podSpec }}
+      {{- toYaml . | nindent 6}}
```
Comment (monotek) [inline zammad/templates/job-init.yaml:39]: might i ask for your usecase and what kind of stuff you want to change?

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by mgruner — COMMENTED** [review]
Looks good to me!

Do you think we could add this to [init-podspec/zammad/ci/full-values.yaml](init-podspec/zammad/ci/full-values.yaml)? This file is used in CI by chart-testing to create a cluster with a lot of possible value customizations to ensure it can still be installed and started properly?

**Comment by klml** [comment]
@mgruner I added [podSpec activeDeadlineSeconds 1337](https://github.com/zammad/zammad-helm/pull/298/commits/bf0e7dce3d56b5720547a6a45f42007f2e7f94e4) or should I use other example values?

**Review by monotek — APPROVED** [review]

File: zammad/ci/full-values.yaml [hunk 3]
```diff
@@ -87,3 +87,6 @@ podLabels:
 
 podAnnotations:
   my-pod-annotation: my-pod-annotation-value
+
+podSpec:
+  activeDeadlineSeconds: 1337
```
Comment (mgruner) [inline zammad/ci/full-values.yaml:92]: I believe this must be added to the `initJob` section, just as in values.yaml.

**Review by mgruner — CHANGES_REQUESTED** [review]

File: zammad/ci/full-values.yaml (hunk 3 above)
Comment (klml) [reply zammad/ci/full-values.yaml]: @mgruner oh yes, you are right, sorry this was my fault.

**Review by mgruner — APPROVED** [review]

**Pull request merged**